packages/blade/src/components/Tour/utils.ts

```diff
@@ -0,0 +1,148 @@
+/* eslint-disable consistent-return */
+/* eslint-disable @typescript-eslint/prefer-nullish-coalescing */
+/* eslint-disable one-var */
+/* eslint-disable prefer-const */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React from 'react';
+import { usePrevious } from '~utils';
+import { throwBladeError } from '~utils/logger';
+import { useScrollLock } from '~utils/useScrollLock';
+
+/**
+ * Hook to delay the state change
+ *
+ * This is used to delay the active step change to allow for transitions to finish
+ * This prevents the popover's footer from changing it's JSX while it's transitioning
+ */
+function useDelayedState<T>(initialState: T, delay: number): T {
+  const [delayedState, setDelayedState] = React.useState(initialState);
+  const timeoutRef = React.useRef<number | undefined>(undefined);
+
+  React.useEffect(() => {
+    timeoutRef.current = window.setTimeout(() => {
+      setDelayedState(initialState);
+    }, delay);
+
+    return () => {
+      window.clearTimeout(timeoutRef.current);
+    };
+  }, [delay, initialState]);
+
+  return delayedState;
+}
+
+/**
+ * Keep track of when we are transitioning between steps
+ *
+ * This is used to prevent the popover from jumping to the next step before animations are finished
+ */
+const useIsTransitioningBetweenSteps = (activeStep: number, transitionDelay: number) => {
+  const prevActiveStep = usePrevious(activeStep);
+  const [isTransitioning, setIsTransitioning] = React.useState(false);
+
+  // Keep track of when we are transitioning between steps
+  React.useEffect(() => {
+    if (prevActiveStep === undefined) return;
+    setIsTransitioning(true);
+    const timeout = setTimeout(() => {
+      setIsTransitioning(false);
+    }, transitionDelay);
+
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [prevActiveStep, transitionDelay]);
+
+  return isTransitioning;
+};
+
+// https://stackoverflow.com/questions/46795955/how-to-know-scroll-to-element-is-done-in-javascript
+function smoothScroll(element: Element | null, options: ScrollIntoViewOptions) {
```

Comment:
Does this have proper browser support? Will the logic break in any older browsers?

Reply:
scrollIntoView is supported in all major [browsers](https://caniuse.com/mdn-api_element_scrollintoview). 

And in older browsers the 'smooth' option might be unavailable but it won't break anything.